samples/kotlin-cli/src/main/kotlin/org/jellyfin/sample/cli/Options.kt

```diff
@@ -3,15 +3,50 @@ package org.jellyfin.sample.cli
 import com.github.ajalt.clikt.core.ParameterHolder
 import com.github.ajalt.clikt.parameters.options.option
 import com.github.ajalt.clikt.parameters.options.required
+import kotlinx.coroutines.runBlocking
+import org.jellyfin.sdk.Jellyfin
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.extensions.authenticateUserByName
+import org.jellyfin.sdk.api.client.extensions.userApi
 
 fun ParameterHolder.serverOption() = option(
 	"-s", "--server",
 	help = "Url of the server",
 	envvar = "JELLYFIN_SERVER"
 ).required()
 
-fun ParameterHolder.tokenOption() = option(
-	"-t", "--token",
-	help = "Access token",
-	envvar = "JELLYFIN_TOKEN"
-).required()
+fun ParameterHolder.apiInstanceHolder(jellyfin: Jellyfin): Lazy<ApiClient> {
+	val server = serverOption().also(::registerOption)
+
+	val token = option(
+		"-t", "--token",
+		help = "Access token",
+		envvar = "JELLYFIN_TOKEN"
+	).also(::registerOption)
+
+	val username = option(
+		"-u", "--username",
+		help = "Username"
+	).also(::registerOption)
+
+	val password = option(
+		"-p", "--password",
+		help = "Password"
+	).also(::registerOption)
+
+	return lazy {
+		runBlocking {
```

Comment:
Not the biggest fan of the runBlocking, but it's only a sample, I suppose? If you don't have a better alternative on hand already, we can leave it like this.

Reply:
All the commands use runBlocking because Clikt is not async